### PR TITLE
Return the exception object from assert_raise

### LINKED ISSUE #5: Capture the exception in `assert_raise`
It would be useful to capture the exception in an `assert_raise` to test the message is the expected one. I'm testing the responses of a library that might return several different errors using the same subclass of `Exception`, and I would like to make sure the `err.message` is the one I should be getting.

Not sure if this is doable with Crystal macros as I haven't really delved into that yet.

### PR DESCRIPTION
**Pull request by foca**
Fixes #5 

Ok, turns out this was trivial 🎉

### CODE REVIEW

File: spec/assertions_spec.cr [hunk 1]
```diff
@@ -43,6 +43,12 @@ describe "Assertions" do
       it "should fail if the code block does not raise any exceptions" do
         assert_raise(Crotest::AssertionFailed) { assert_raise { } }
       end
+
+      it "returns the exception object" do
+        ex = assert_raise { raise "Boom!" }
+        assert_equal Exception, ex.class
+        assert_equal "Boom!", ex.to_s
+      end
```
Comment (emancu) [inline spec/assertions_spec.cr]: Probably, you want to test both cases, when you specify the Exception class and using the default.

File: src/crotest/assertions.cr [hunk 2]
```diff
@@ -33,6 +33,7 @@ module Crotest::Assertions
       {{yield}}
     rescue %exception : {{expected}}
       Crotest.increment(:assertions)
+      %exception
     rescue %exception
       %result = %exception.is_a?({{expected}})
```
Comment (emancu) [inline src/crotest/assertions.cr]: I guess you need to return the `%exception` in this case too

File: spec/assertions_spec.cr (hunk 1 above)
Comment (foca) [inline spec/assertions_spec.cr]: Ah, I was lazy, it felt like duplicating the test, but I'm ok adding it.

File: src/crotest/assertions.cr (hunk 2 above)
Comment (foca) [inline src/crotest/assertions.cr]: True! I sort of overlooked this assuming this was the failure case. Will add.

File: spec/assertions_spec.cr (hunk 1 above)
Comment (emancu) [inline spec/assertions_spec.cr]: Could you use the `DummyException` here too and change the code to `assert_raise DummyException { raise "Boom!" }`

In that way, we cover both cases explicitly

**Pull request merged**